src/libexpr/eval.cc

```diff
@@ -2296,6 +2317,75 @@ StorePath EvalState::coerceToStorePath(const PosIdx pos, Value & v, NixStringCon
     error("path '%1%' is not in the Nix store", path).withTrace(pos, errorCtx).debugThrow<EvalError>();
 }
 
+std::pair<DerivedPath, std::string_view> EvalState::coerceToDerivedPathUnchecked(const PosIdx pos, Value & v, std::string_view errorCtx)
+{
+    NixStringContext context;
+    auto s = forceString(v, context, pos, errorCtx);
+    if (context.size() != 1)
+        error("string '%s' has multiple entries in its context. It should only have one entry", s)
```

Comment:
Error reads a bit odd when the supplied string has zero context.

Reply:
Now has a number instead, and we test for both 0 and 2